cmake/Stdlib.cmake

```diff
@@ -1,33 +1,229 @@
 #
-#  Copyright (c) 2018-2023, Intel Corporation
+#  Copyright (c) 2018-2024, Intel Corporation
 #
 #  SPDX-License-Identifier: BSD-3-Clause
 
 #
 # ispc Stdlib.cmake
 #
-function(create_stdlib mask outputPath)
-    set(output ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/stdlib_mask${mask}_ispc.cpp)
+
+function (write_stdlib_bitcode_lib name target os bit)
+    set(arch "error")
+    if ("${target}" MATCHES "sse|avx")
+        if ("${bit}" STREQUAL "32")
+            set(arch "x86")
+        elseif ("${bit}" STREQUAL "64")
+            set(arch "x86_64")
+        else()
+            set(arch "error")
+        endif()
+    elseif ("${target}" MATCHES "neon")
+        if ("${bit}" STREQUAL "32")
+            set(arch "arm")
+        elseif ("${bit}" STREQUAL "64")
+            set(arch "aarch64")
+        else()
+            set(arch "error")
+        endif()
+    elseif ("${target}" MATCHES "wasm")
+        if ("${bit}" STREQUAL "32")
+            set(arch "wasm32")
+        elseif ("${bit}" STREQUAL "64")
+            set(arch "wasm64")
+        else()
+            set(arch "error")
+        endif()
+    elseif ("${target}" MATCHES "gen9|xe")
+        set(arch "xe64")
+    endif()
+    if ("${arch}" STREQUAL "error")
+        message(FATAL_ERROR "Incorrect target or bit passed to write_stdlib_bitcode_lib ${target} ${os} ${bit}")
+    endif()
+
+    if ("${os}" STREQUAL "unix")
+        set(os "linux")
+    endif()
```

Comment:
This part is a copy from GenerateBuiltins.cmake. I'd like to avoid logic duplication about ISPC's matrix support as much as possible. Let's create a separata Cmake module that will keep this logic and extract this part there into a separate function:
```
function(detect_architecture target bit os arch)
...
 set(${arch} ${${arch}} PARENT_SCOPE)
    set(${os} ${${os}} PARENT_SCOPE)
endfunction()

function(write_target_bitcode_lib name target os bit)
    detect_architecture(${target} ${bit} ${os} arch)
...
endfunction()
```

Reply:
I created a file with common piece of code between builtins and stdlib cmakes.